### DISCUSSION TITLE
awswrangler copy_objects

### DISCUSSION THREAD

**Discussion by nmakarun** [aws/aws-sdk-pandas#2003] [General]
I am copying objects from one prefix to another using awswrangler and it works perfectly fine, but I need to get rid of the source prefix when I copy to target and keep just the Key name from source prefix.

wr.s3.copy_objects(full_paths_list, source_path, target_path)

Any idea on how can I achieve this, I looked at the documentation (https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.s3.copy_objects.html)

Couldnt find anything tangible.

**Comment by malachi-constant** [comment]
Is this what you are trying to do?

```python
>>> wr.s3.copy_objects(['s3://hansonlu-test-data-bucket/source/extra-prefix/file1.json'], 's3://hansonlu-test-data-bucket/source/extra-prefix/', 's3://hansonlu-test-data-bucket/target/')
['s3://hansonlu-test-data-bucket/target/file1.json']
```

**Reply by nmakarun** [reply]
Yes, this is what I am trying to do. But when I perform it the target looks like

s3://hansonlu-test-data-bucket/target/hansonlu-test-data-bucket/source/extra-prefix/file1.json

but I want s3://hansonlu-test-data-bucket/target/file1.json

**Reply by malachi-constant** [reply]
I just ran the above on v`2.19.0` do you happen to be on an older version of `awswrangler`?

**Reply by nmakarun** [reply]
Thanks for this, I am using 2.15.1 version in glue version 3.0, Can I update this by using --update parameter or do I need to use a .whl file by downloading in S3.

**Reply by malachi-constant** [reply]
You should be able to [update a Glue Job's version of the library](https://aws-sdk-pandas.readthedocs.io/en/stable/install.html#aws-glue-pyspark-jobs) by adding the version number in ```--additional-python-modules```.

**Reply by nmakarun** [reply]
Thanks, That worked but interestingly. It doesnt work, when I pass the source and target s3 paths as variables, but works when I pass the hard coded values
E.g.
    source_path = f"{bucket_uri}/{source_prefix}/"
    target_path = f"{bucket_uri}/{new_prefix}/"
   wr.s3.copy_objects(full_paths_list, source_path, target_path)
Above scenario doesnt work.

E.g.
wr.s3.copy_objects(full_paths_list, 's3://bucket-name/source/', 's3://bucket-name/target/')

Above case works, not sure if this is the expected behaviour.

**Reply by nmakarun** [reply]
Found the issue, it works when I use single quotes with format method but not with f string formatting method

    source = 's3://{}/source'.format('bucketname')
    target = 's3://{}/target'.format('bucketname')
    wr.s3.copy_objects(full_paths_list, source , target)

this works.

**Reply by malachi-constant** [reply]
Hmm.. f-strings should work the same way. Here's a working example.

```python
>>> import awswrangler as wr
>>> bucket = "hansonlu-test-data-bucket"
>>> source_path = f"s3://{bucket}/source/extra-prefix/"
>>> target_path = f"s3://{bucket}/target/"
>>> paths_list = [f"{source_path}file1.json"]
>>> wr.s3.copy_objects(paths_list, source_path, target_path)
['s3://hansonlu-test-data-bucket/target/file1.json']
```
